File: backend-frontend/src/app/interfaces/Alert.ts
import { Transaction } from "./Transaction";
import { RefObject } from "react";

export interface Alert {
  alertId: string;
  ruleId: number;
  rulePayload: string;
  triggeringValue: number;
  triggeringEvent: Transaction;
  ref: RefObject<HTMLDivElement>;
}

File: backend-frontend/src/main/java/com/ververica/demo/backend/model/Alert.java
package com.ververica.demo.backend.model;

import com.ververica.demo.backend.datasource.Transaction;
import java.math.BigDecimal;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class Alert {
  private Integer ruleId;
  private String rulePayload;

  Transaction triggeringEvent;
  BigDecimal triggeringValue;
}
